### PR TITLE
hal/imxrt106x: init enet PLL

### PR DESCRIPTION
**Pull request by julianuziemblo**
## Description
- fix clk0 enable bit of enet PLL
- add bypass during clock frequency change
- initialize enet PLL
- remove enet PLL deinitialization

## Motivation and Context
enet driver for imxrt106x-evk needs this 

Related: [https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/601](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/601) and [https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/595](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/595)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `armv7a7-imx6ull-evk`, `armv7m7-imxrt106x-evk`.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/imxrt/10xx/106x/imxrt.c [hunk 1]
```diff
@@ -1659,7 +1660,7 @@ void _imxrt_init(void)
 
 	/* Disable unused clocks */
 	*(imxrt_common.ccm + ccm_ccgr0) = 0x00c0ffff;
-	*(imxrt_common.ccm + ccm_ccgr1) = 0x300c0000;
+	*(imxrt_common.ccm + ccm_ccgr1) = 0x300c0c00;
```
Comment (nalajcie) [inline hal/armv7m/imxrt/10xx/106x/imxrt.c:1663]: no, this needs to be done in ethernet driver using platformctl on startup. If we're not running lwip, we don't want to have this clock enabled.

**Comment by nalajcie** [comment]
the code looks good, please fix commit message:
 - the change is done only for `imxrt106x`, so please use it as a prefix
 - IMHO it's worth to describe WHY bypass enable is needed - either in the code as a comment and/or in the commit message body - eg something like `/* enable bypass during clock frequency change */`

File: hal/armv7m/imxrt/10xx/106x/imxrt.c [hunk 2]
```diff
@@ -683,10 +683,14 @@ void _imxrt_ccmInitEnetPll(u8 enclk0, u8 enclk1, u8 enclk2, u8 div0, u8 div1)
 		enet_pll |= 1 << 21;
 	}
 
-	*(imxrt_common.ccm_analog + ccm_analog_pll_enet) = enet_pll;
+	*(imxrt_common.ccm_analog + ccm_analog_pll_enet) = 1 << 16; /* enable bypass during clock frequency change */
```
Comment (nalajcie) [inline hal/armv7m/imxrt/10xx/106x/imxrt.c:686]: personally I would prefer to have the comment in the line before (as the comment is for the whole code block until line 693 - and not to create too-long lines unnecessarily), but I'm not enforcing it.

**Review by nalajcie — APPROVED** [review]

File: hal/armv7m/imxrt/10xx/106x/imxrt.c (hunk 2 above)
Comment (julianuziemblo) [reply hal/armv7m/imxrt/10xx/106x/imxrt.c]: good idea for clarity, will apply it

**Review by nalajcie — APPROVED** [review]

**Pull request merged**